### ISSUE TITLE
Extract website icon resources via WebExtension

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#2080]
* See #1791 about components that make use of WebExtensions.
* GeckoView: https://bugzilla.mozilla.org/show_bug.cgi?id=1527618

Currently the GeckoView prefers to not add any icon related APIs. Instead the suggestion is to use the WebExtension API [(Bug 1518841)](https://bugzilla.mozilla.org/show_bug.cgi?id=1518841) and the Messaging API [(Bug 1518843)](https://bugzilla.mozilla.org/show_bug.cgi?id=1518843) to add this functionality.

For our icon pipeline we'd need:
* Favicon URL(s)
* Touch icon URL(s)
* Potentially other icons linked in markup.
* In addition to the URLs we also use (if in markup): size and MIME type.

**Comment by pocmo** [comment]
Depends on #2243

**Comment by cpeterso** [comment]
I closed GV's favicon bug as WONTFIX because @snorp says A-C can fetch favicons themselves without a GV API.

https://bugzilla.mozilla.org/show_bug.cgi?id=1527618

**Comment by jhugman** [comment]
Firefox iOS replaced a lot of handwritten JS with [@mozilla/page-metadata-parser][1] with great success.

I'd be interested in doing the same here.

[1]: https://github.com/mozilla/page-metadata-parser

**Comment by cpeterso** [comment]
@pocmo You added the `blocked` label to this issue back in March. What is this issue blocked on? GV WebExtension support? We decided in GV bug https://bugzilla.mozilla.org/show_bug.cgi?id=1527618 that GV would not (need to build) a GV API to share website icons because AC could write a WebExtension.

**Comment by pocmo** [comment]
@cpeterso This was waiting for the messaging API in GV/AC. Not blocked anymore.
